PracticeGit/conversion.c

```diff
@@ -0,0 +1,125 @@
+#pragma warning(disable:4996)
+#include <stdio.h>
+#include <limits.h>
+
+#define BITS 32
+#define HEX_LENGTH 8
+
+int binary[BITS];   // Global array to hold binary representation
+
+/// <summary>Function to convert decimal to Binary.</summary>
+void DecToBinary (int n) {
+   for (int i = 0; i < BITS; i++)  binary[i] = (n >> i) & 1;
+}
+
+/// <summary>Print the binary representation.</summary>
+void PrintBinary (int n) {
+   printf ("Binary Equivalent for %d: ", n);
+   for (int i = BITS - 1; i >= 0; i--) {
+      printf ("%d", binary[i]);
+      if ((i+1) % 8 == 0 && i != BITS - 1) printf (" ");
+   }
+   printf ("\n");
+}
+
+/// <summary>Convert a value to its corresponding hexadecimal character.</summary>
+char ValueToHexChar (int value) {
+   return (value < 10) ? (char)(value + '0') : (char)(value - 10 + 'A');
+}
+
+/// <summary>Function to convert decimal to Hexadecimal.</summary>
+void DecToHexadecimal (char hexString[HEX_LENGTH + 1]) {
+   //char hexDigits[8];  // Array to store hexadecimal digits
+   int hexIndex = 0;
+   for (int i = BITS - 1; i >= 0; i -= 4) {  // Convert binary array to hexadecimal
+      int value = 0;
+      for (int j = 0; j < 4; j++) value = (value << 1) | binary[i - j];
+      hexString[hexIndex++] = ValueToHexChar (value);
+   }
+   while (hexIndex < 8) hexString[hexIndex++] = '0';  // Fill remaining digits with '0'
+   hexString[HEX_LENGTH] = '\0';  // Null-terminate the string
+}
+
+/// <summary>Print the hexadecimal string.</summary> 
+void PrintHexadecimal (int n, const char hexString[HEX_LENGTH]) {
+   printf ("Hexadecimal Equivalent for %d: %s\n", n, hexString);
+}
+
+/// <summary> Function to run test cases.</summary>
+void RunTestCases () {
+   struct TestCase {
+      long long input;
+      const char* expectedBinary;
+      const char* expectedHexadecimal;
+   };
+   struct TestCase testCases[] = {
+       {42, "00000000 00000000 00000000 00101010", "002A"},
+       {2147483648, "01111111 11111111 11111111 11111111", "7FFFFFFF"},
+       {-2147483647, "10000000 00000000 00000000 00000000", "80000000"},
+       {0, "00000000 00000000 00000000 00000000", "00000000"},
+       {-123, "11111111 11111111 11111111 10000101", "FFFFFF85"},
+       {255, "00000000 00000000 00000000 11111111", "000000FF"},
+       {100, "00000000 00000000 00000000 01100100", "00000064"},
+   };
+   char hexString[HEX_LENGTH + 1];
+   printf ("Running Test Cases:\n");
+   for (int i = 0; i < sizeof (testCases) / sizeof (testCases[0]); i++) {
+      long long inputNumber = testCases[i].input;
+      int decimalNumber;
+      printf ("Testing input: %lld\n", inputNumber);
+      if (inputNumber < INT_MIN || inputNumber > INT_MAX) {
+         printf ("Value out of range for a 32-bit signed integer.\n");
+         continue;
+      }
+      else {
+         decimalNumber = (int)inputNumber;
+         DecToBinary (decimalNumber);

```

Comment:
The output and the test cases have to be checked if they are equal. here you are just printing the output and test cases. Please check it.